test/foundry-tests/lib/FeeEscalatorLib.t.sol

```diff
@@ -0,0 +1,124 @@
+        assertEq(FeeEscalatorLib.resolve(1 ether, 2 ether, 100, 200), 2 ether);
+    }
+
+    function testRelayFeeEscalationRevertsWithWrongEndStartTimes() public {
+        vm.expectRevert(FeeEscalatorLib.EndTimeBeforeStartTime.selector);
+        FeeEscalatorLib.resolve(1 ether, 2 ether, 200, 100);
+    }
+
+    function testRelayFeeEscalationEqualAmounts(uint256 amount, uint256 startTime, uint256 endTime) public {
+        vm.assume(endTime >= startTime);
+        uint256 time = startTime;
+        bound(time, startTime, startTime);
+
+        vm.warp(time);
+        assertEq(FeeEscalatorLib.resolve(amount, amount, startTime, endTime), amount);
+    }
+
+    function testRelayFeeEscalationInvalidAmounts() public {
+        vm.expectRevert(FeeEscalatorLib.InvalidAmounts.selector);
+        FeeEscalatorLib.resolve(2 ether, 1 ether, 100, 200);
+    }
+
+    function testRelayFeeEscalationBounded(uint256 startAmount, uint256 endAmount, uint256 startTime, uint256 endTime)
+        public
+    {
+        vm.assume(endAmount > startAmount);
+        vm.assume(endTime >= startTime);
+        uint256 resolved = FeeEscalatorLib.resolve(startAmount, endAmount, startTime, endTime);
+        assertGe(resolved, startAmount);
+        assertLe(resolved, endAmount);
+    }
+
+    function testToTokenPermissions() public {
+        address token = makeAddr("token");
+        FeeEscalator memory fee = FeeEscalator({
+            token: token,
+            startAmount: 1 ether,
+            endAmount: 2 ether,
+            startTime: 100,
+            endTime: 200,
+            recipient: address(0)
+        });
+        ISignatureTransfer.TokenPermissions memory permission = fee.toTokenPermissions();
+        assertEq(permission.token, token);
+        // should be endAmount
+        assertEq(permission.amount, 2 ether);
+    }
+
+    function testToTransferDetailsWithSpecifiedRecipient() public {
+        address filler = makeAddr("filler");
+        FeeEscalator memory fee = FeeEscalator({
+            token: address(this),
+            startAmount: 1 ether,
+            endAmount: 1 ether,
+            startTime: 0,
+            endTime: 0,
+            recipient: address(this)
+        });
+        ISignatureTransfer.SignatureTransferDetails memory details = fee.toTransferDetails(filler);
+        assertEq(details.to, address(this));
+        assertEq(details.requestedAmount, 1 ether);
+    }
+
+    function testToTransferDetailsWithNoRecipient() public {
+        address filler = makeAddr("filler");
+        FeeEscalator memory fee = FeeEscalator({
+            token: address(this),
+            startAmount: 1 ether,
+            endAmount: 1 ether,
+            startTime: 0,
+            endTime: 0,
+            recipient: address(0)
+        });
+        ISignatureTransfer.SignatureTransferDetails memory details = fee.toTransferDetails(filler);
+        assertEq(details.to, filler);
+        assertEq(details.requestedAmount, 1 ether);
+    }
```

Comment:
Note that I would like to use diff naming conventions but I can do that in followup

Reply:
yep sg, I was gonna change here but lets do it all at once